pages/product.js

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Error from 'next/error';
+import Bookmark from '../components/Icons/Bookmark';
+import QuantityButton from '../components/QuantityButton';
+import Breadcrumb from '../components/Breadcrumb';
+import Carousel from '../components/Carousel';
+import ProductInfo from '../components/ProductInfo';
+import { withRouter } from 'next/router';
+import fetch from 'isomorphic-unfetch';
+import { server } from '../server/config';
+
+/**
+ * Product Details Page
+ *
+ * @export
+ * @class ProductPage
+ * @extends {React.PureComponent}
+ */
+class ProductPage extends React.PureComponent {
+  state = {
+    qty: 0,
+  };
+
+  static propTypes = {
+    Item: PropTypes.shape({
+      id: PropTypes.number,
+      name: PropTypes.string,
+      imageUrl: PropTypes.string,
+      originalPrice: PropTypes.number,
+      discountedPrice: PropTypes.number,
+    }),
+  };
+
+  static getInitialProps = async ({ query, res }) => {
+    const { id } = query;
+    if (!id && res) {
+      res.statusCode = 404;
+      return { Item: null };
+    }
+    const { data: Item } = await fetch(`${server}/api/items/${id}`).then(r =>
+      r.json()
+    );
+    return { Item };
+  };
+  increment = () => {
+    this.setState({
+      qty: this.state.qty + 1,
+    });
+  };
+  decrement = () => {
+    this.setState({
+      qty: this.state.qty - 1,
```

Comment:
Add check for -ve

Reply:
@bogas04 The `qty` won't be negative because I have `disabled` the decrement button once the value becomes zero. I shall add `cursor: not-allowed` so that user can understand it.